pkg/backporter/backporter.go

```diff
@@ -278,94 +304,218 @@ func GetBugHandler(client bugzilla.Client) HandlerFuncWithErrorReturn {
+	bugIDStr := req.URL.Query().Get(BugIDQuery)
+	if bugIDStr == "" {
+		w.WriteHeader(http.StatusBadRequest)
+		wpErr := writePage(w, "Error.", errorTemplate, fmt.Sprintf("%s - query incorrect", BugIDQuery))
+		if wpErr != nil {
+			http.Error(w, "Error building page.", http.StatusInternalServerError)
+			return wpErr
+		}
+		return fmt.Errorf("%s - query incorrect", BugIDQuery)
+	}
+	bugID, err := strconv.Atoi(bugIDStr)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		wpErr := writePage(w, "Error.", errorTemplate, fmt.Sprintf("Unable to parse bug id: %v", err))
+		if wpErr != nil {
+			http.Error(w, "Error building page.", http.StatusInternalServerError)
+			return wpErr
+		}
+		return err
+	}
+
+	wrpr, err := getClonesTemplateData(bugID, w, client, allTargetVersions)
+	if err != nil {
+		return err
+	}
+	wpErr := writePage(w, "Clones", clonesTemplate, wrpr)
+	if wpErr != nil {
+		http.Error(w, "Error building page.", http.StatusInternalServerError)
+		return wpErr
+	}
+	return nil
+}
+
+// CreateCloneHandler will create a clone of the specified ID and return success/error
+func CreateCloneHandler(client bugzilla.Client, allTargetVersions sets.String) HandlerFuncWithErrorReturn {
 	return func(w http.ResponseWriter, req *http.Request) error {
-		bugIDStr := req.URL.Query().Get(BugIDQuery)
-		if bugIDStr == "" {
+		if req.Method != "POST" {
 			w.WriteHeader(http.StatusBadRequest)
-			wpErr := writePage(w, "Error!", emptyTemplate, fmt.Sprintf("%s - query incorrect", BugIDQuery))
+			wpErr := writePage(w, "Error.", errorTemplate, fmt.Sprintf("%s - query incorrect", BugIDQuery))
 			if wpErr != nil {
-				http.Error(w, "Error building page!", http.StatusInternalServerError)
+				http.Error(w, "Error building page.", http.StatusInternalServerError)
 				return wpErr
 			}
-			return fmt.Errorf("%s - query incorrect", BugIDQuery)
+			return fmt.Errorf("Bad request - clones/create can handle only POST requests")
 		}
-		bugID, err := strconv.Atoi(bugIDStr)
+		// Parse the parameters passed in the POST request
+		err := req.ParseForm()
 		if err != nil {
 			w.WriteHeader(http.StatusBadRequest)
-			wpErr := writePage(w, "Error!", emptyTemplate, fmt.Sprintf("Unable to parse bug id: %v", err))
+			wpErr := writePage(w, "Error", errorTemplate, fmt.Sprintf("Unable to parse request : %v", err))
 			if wpErr != nil {
-				http.Error(w, "Error building page!", http.StatusInternalServerError)
+				http.Error(w, "Error building page.", http.StatusInternalServerError)
 				return wpErr
 			}
 			return err
 		}
-		bug, err := client.GetBug(bugID)
+		bugID, err := strconv.Atoi(req.FormValue("ID"))
 		if err != nil {
-			w.WriteHeader(http.StatusNotFound)
-			wpErr := writePage(w, "Not Found", emptyTemplate, fmt.Sprintf("Bug#%d not found", bugID))
+			w.WriteHeader(http.StatusBadRequest)
+			wpErr := writePage(w, "Error", errorTemplate, fmt.Sprintf("Invalid bug id. - Bug#%d : %v", bugID, err))
```

Comment:
Please be helpful and tell how it was invalid. Here: Could not parse string as int. Which could be because it wasn't passed as query arg in which case this is even more confusing so maybe it would be a good idea to have a different error in case it wasn't sent